components/clp-rust-utils/src/serde/yaml_file.rs

```diff
@@ -0,0 +1,23 @@
+use serde::de::DeserializeOwned;
+
+use crate::Error;
+
+/// Reads a YAML configuration file from a specified path and deserializes its contents
+/// into the requested type `T`.
```

Comment:
```suggestion
/// Reads a YAML file from a specified path and deserializes its contents into the requested type
/// `T`.
```
It doesn't necessarily to be a config file, right?